content/6_Advanced/Count_Min.mdx

```diff
@@ -3,38 +3,219 @@ id: count-min
+Node query(int l, int r) {
+	Node res;
+	for (l += MAXN, r += MAXN + 1; l < r; l >>= 1, r >>= 1) {
+		if (l & 1) res = merge(res, T[l++]);
+		if (r & 1) res = merge(res, T[--r]);
+	}
+	return res;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+### Solution - Area of Rectangles
+
+<Spoiler title = "Hint 1">
+
+Count the number of squares that are covered by **none** of the input rectangles.
+
+</Spoiler>
+
+<Spoiler title = "Hint 2">
+
+Sort the rectangles by x-coordinate and run a sweep-line.
 
 </Spoiler>
 
 <Spoiler title="Solution">
 
-<IncompleteSection>
+We can use techniques introduced in [Range Queries with Sweep Line](https://usaco.guide/plat/range-sweep?lang=cpp)
+
+We sweep from left to right over the $x$-coordinates, maintaining two events for each rectangle: one for the left boundary and one for the right boundary. Maintain a Lazy Segment Tree over the $y$-coordinates. 
+- When we run into a left boundary of some rectangle with $y$-coordinates $(y_0, y_1)$, increase each index $i \in [y_0, y_1]$ by 1
+- When we run into a right boundary of some rectangle with $y$-coordinates $(y_0, y_1)$, decrease each index $i \in [y_0, y_1]$ by 1
+
+Then, for each $x$, we simply need to count the number of non-zero indices which corresponds to indices that are covered by at least one rectangle. How can we do this?
 
-describe
+Instead of counting the area covered by at least one rectangle, let's count the amount of space covered by no rectangles. We can subtract this amount from the total number of indices to get the value we want. 
 
-</IncompleteSection>
+We can use a Segment Tree that counts the number of occurences of the minimum value. Because the minimum value is at least zero (there can't be a negative number of rectangles at a position) the number of uncovered squares is equal to the number of squares with value 0.
 
 </Spoiler>
 
+### Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAXX = 1e6;
+const int MAXN = 2 * MAXX + 1;
+ 
+struct Event {
+	int t, x, y0, y1;
+	// t = 1 for left bound, -1 for right bound
+	bool operator<(const Event &e) { return x < e.x; }
+};
+ 
+int N;
+vector<Event> E;
+	
+// segment tree
+pair<long long, long long> T[MAXN * 4];
+int L[MAXN * 4];
+ 
+pair<long long, long long> merge(pair<long long, long long> a, pair<long long, long long> b) {
+	if (a.first < b.first)
+		return a;
+	if (a.first > b.first)
+		return b;
+	return MP(a.first, a.second + b.second);
+}
+ 
+void pushdown(int t) {
+	if (L[t]) {
+		T[t << 1].first += L[t];
+		L[t << 1] += L[t];
+		T[t << 1 | 1].first += L[t];
+		L[t << 1 | 1] += L[t];
+		L[t] = 0;
+	}
+}
+ 
+void build(int t = 1, int tl = 0, int tr = MAXN) {
+	if (tl == tr) {
+		T[t] = MP(0, 1);
+		return;
+	}
+	int tm = (tl + tr) >> 1;
+	build(t << 1, tl, tm);
+	build(t << 1 | 1, tm + 1, tr);
+	T[t] = merge(T[t << 1], T[t << 1 | 1]);
+}
+ 
+void update(int l, int r, int v, int t = 1, int tl = 0, int tr = MAXN) {
```

Comment:
tell the user what these updates and queries actually mean maybe
(also pushdown what)

Reply:
these are regular lazy segtree function?

Reply:
i have never coded a single pushdown in my life

Reply:
bruh